frontend/src/models/groupsModel.ts

```diff
@@ -0,0 +1,29 @@
+import { kea } from 'kea'
+import api from 'lib/api'
+import { GroupType } from '~/types'
+import { teamLogic } from 'scenes/teamLogic'
+import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
+import { FEATURE_FLAGS } from 'lib/constants'
+import { groupsModelType } from './groupsModelType'
+
+export const groupsModel = kea<groupsModelType>({
+    connect: {
+        values: [teamLogic, ['currentTeamId']],
+    },
+    loaders: ({ values }) => ({
+        groupTypes: [
+            [] as Array<GroupType>,
+            {
+                loadAllGroupTypes: async () => {
+                    if (featureFlagLogic.values.featureFlags[FEATURE_FLAGS.GROUP_ANALYTICS]) {
```

Comment:
So we should also check whether clickhouse is enabled here.

Suggestion: Extract a selector for the feature flag check + checking whether clickhouse is enabled, use it elsewhere around here.